src/backwardproblem.cpp

```diff
@@ -59,44 +65,55 @@ void BackwardProblem::workBackwardProblem() {
     }
 
     int it = model->nt() - 1;
-    model->initializeB(xB, dxB, xQB);
-    handleDataPointB(it);
-    solver->setupB(&which, model->getTimepoint(it), model, xB, dxB, xQB);
+    /* If we have posteq, infty timepoints were already treated */
+    for (int jt = model->nt() - 1; jt >= 0; jt--)
+        if (std::isinf(model->getTimepoint(it)))
+            --it;
+    if (it < model->nt() - 1){
```

Comment:
cant we just initialize `xQB` to zeros for `it < model->nt() - 1` and pass it to `initializeB`? I think that would be better readable. Also shouldn't if/else cases be switched here?

Reply:
The problem ist that, if we have postequilibration (which is the case for `it < model->nt() - 1`), xQB already contains the quadratures from backward postequilibration. Hence, in that case, we don't want to overwrite them, but still want to reinitialize xB and dxB.
I have another idea, which should simplify this. Will commit.